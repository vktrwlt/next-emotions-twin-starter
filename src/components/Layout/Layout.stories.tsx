import React from 'react';
import { Story } from '@storybook/react/types-6-0';
import { Layout, LayoutProps } from '@/components/Layout/Layout';

export default {
  title: 'Layout',
  component: Layout,
};

const Template: Story<LayoutProps> = (args) => <Layout {...args} />;

export const Default = Template.bind({});
Default.args = { variant: 'primary', size: 'md' };
