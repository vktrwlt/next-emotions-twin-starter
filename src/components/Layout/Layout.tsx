import { Header } from '@/components/Header';
import { Footer } from '@/components/Footer';
import tw, { styled, css, theme } from 'twin.macro';

export interface LayoutProps {
  children: React.ReactNode;
}

const Box = styled.div`
  ${tw`flex flex-col min-h-screen overflow-x-hidden bg-white`}
`;
const ContentWrapper = styled.div`
  ${tw`flex-auto w-full mt-md`}
`;

const Layout = (props: LayoutProps) => {
  const { children } = props;

  return (
    <Box>
      <Header />
      <ContentWrapper>
        <main>{children}</main>
      </ContentWrapper>
      <Footer />
    </Box>
  );
};

export { Layout };
