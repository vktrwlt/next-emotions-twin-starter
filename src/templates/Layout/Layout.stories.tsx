import React from 'react';
import { Story } from '@storybook/react/types-6-0';
import { Layout, LayoutProps } from '@/templates/Layout/Layout';

export default {
  title: 'Layout',
  component: Layout,
};

const Template: Story<LayoutProps> = (args) => <Layout {...args} />;

export const Default = Template.bind({});
Default.args = { variant: 'primary', size: 'md' };
