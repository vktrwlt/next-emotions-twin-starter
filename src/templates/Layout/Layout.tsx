import { Header } from '@/templates/Layout/Header';
import { Footer } from '@/templates/Layout/Footer';
import tw, { styled, css, theme } from 'twin.macro';

export interface LayoutProps {
  children: React.ReactNode;
}

const Box = styled.div`
  ${tw`flex flex-col min-h-screen overflow-x-hidden bg-white`}
`;
const ContentWrapper = styled.div`
  ${tw`flex-auto w-full mt-md`}
`;

export const Layout = (props: LayoutProps) => {
  const { children } = props;

  return (
    <Box>
      <Header />
      <ContentWrapper>
        <main>{children}</main>
      </ContentWrapper>
      <Footer />
    </Box>
  );
};
