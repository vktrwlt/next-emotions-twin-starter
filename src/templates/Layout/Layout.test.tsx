import { Layout } from './Layout';
import { render, fireEvent } from '@/utils/test-utils';

describe('<Layout />', () => {
  it('should render without crashing', () => {
    const { container } = render(<Layout>Layout</Layout>);
    expect(container).toBeInTheDocument();
  });
});
