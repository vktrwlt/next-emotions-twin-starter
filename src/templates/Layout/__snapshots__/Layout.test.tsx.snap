// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  overflow-x: hidden;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
}

.emotion-4 {
  max-width: 1280px;
  padding-left: 2rem;
  padding-right: 2rem;
  margin-left: auto;
  margin-right: auto;
}

.emotion-6 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  width: 100%;
  margin-top: 24px;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        Header
      </div>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <main>
        Hello
      </main>
    </div>
    <div
      class="emotion-2 emotion-9"
    >
      <div
        class="emotion-4 emotion-5"
      >
        Footer
      </div>
    </div>
  </div>
</DocumentFragment>
`;
